x-pack/plugins/observability_solution/infra/public/components/asset_details/tabs/dashboards/actions/link_dashboard.tsx

```diff
@@ -18,11 +18,13 @@ export function LinkDashboard({
   newDashboardButton = false,
   customDashboards,
   assetType,
+  canLinkOrEditCustomDashboard,
 }: {
   onRefresh: () => void;
   newDashboardButton?: boolean;
   customDashboards?: DashboardItemWithTitle[];
   assetType: InfraCustomDashboardAssetType;
+  canLinkOrEditCustomDashboard: boolean;
```

Comment:
Since we have the `useCanManipulateCustomDashboard`, what if we use it in the components used for the edit, link, and delete operations, instead of passing the flag in the props?

Reply:
I was about to comment on the same thing. I think each of the components mentioned by @crespocarlos here ☝🏻 must get the permission they care about and adapt the UI. It should not be the consumer's responsibility to pass it down.

Reply:
I changed that, thanks. My idea was mostly to be future-proof in case there are other conditions we want to add and pass from the parent but probably that won't be the case.

Reply:
I also updated the tests to mock the hook response as well as now it is called inside the component